HPCC-16377 Display file history information

miguelvazq opened the pull request:
Add the ability to view file history information and remove file history information.

Fixes: HPCC-16377
Fixes: HPCC-17598

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

miguelvazq commented:
@GordonSmith Please review.

miguelvazq commented:
@rpastrana Please review Spanish translations.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 40a306c7d0c2c39d6352451bf31674c51bc70b19
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 94.149 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |724 |724 |0 |
|test (thor) |619 |619 |0 |
|test (roxie) |747 |747 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana reviewed (commented):
@miguelvazq spanish looks fine

richardkchapman commented:
@GordonSmith Back to you

GordonSmith commented on esp/src/eclwatch/FileHistoryWidget.js:
```diff
@@ -0,0 +1,119 @@
+], function (declare, lang, i18n, nlsHPCC, arrayUtil, on, dom, domConstruct, domClass,
+                registry, ToolbarSeparator, Button,
+                selector, ColumnResizer,
+                GridDetailsWidget, WsDfu, DelayLoadWidget, ESPUtil) {
```
Not used:
* on
* dom
* domConstruct
* domClass
* ToolbarSeparator
* Button
* DelayLoadWidget

GordonSmith commented on esp/src/eclwatch/templates/FileHistoryWidget.html:
```diff
@@ -0,0 +1,8 @@
+            </div>
+        </div>
+    </div>
+</div>
```
I don't think this file is used anywhere?

richardkchapman commented:
@miguelvazq

rpastrana commented:
This pull request is dependent on https://github.com/hpcc-systems/HPCC-Platform/pull/9865.
Interface changes requested to 9865 will affect this PR
@miguelvazq FYI

Commit:
HPCC-16377 Display file history information

Add the ability to view file history information and remove file history information.

Fixes: HPCC-16377
Fixes: HPCC-17598

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

miguelvazq commented:
@GordonSmith Waited for blocker to be merged and finalized to make sure front-end had correct response qualifiers. Also, updated as per your comments.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 8dd51a98019b0a148b25dc165328167ceb0feb47
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.906 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |727 |727 |0 |
|test (thor) |620 |620 |0 |
|test (roxie) |748 |748 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/FileHistoryWidget.js:
```diff
@@ -0,0 +1,111 @@
+    return declare("FileHistoryWidget", [GridDetailsWidget], {
+        i18n: nlsHPCC,
+        gridTitle: nlsHPCC.History,
+        idProperty: "name",
```
Should this be "Name" rather than "name"?  
Also is "Name" unique enough?
(probably doesn't matter as you don't have any actions dependent on selection?).

miguelvazq replied:
You are correct it went from name to Name and I missed this one. "Name" is the name refers to the name of the logical file so it should be sufficient.

GordonSmith commented on esp/src/eclwatch/FileHistoryWidget.js:
```diff
@@ -0,0 +1,111 @@
+                        width: 10,
+                        selectorType: 'checkbox',
+                        label: ""
+                    }),
```
Is selection needed?

miguelvazq replied:
Leaving in as we have another method that we may need to discuss if we want to expose to the U/I.

miguelvazq replied:
Removed selection until HPCC-17599 is ready.

GordonSmith reviewed (approved):
Looks ok - minor comments about the need for selection and uniqueID

richardkchapman commented:
@miguelvazq

miguelvazq commented:
@GordonSmith updated.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 7de7db5ba9fb40e28e5b86b3ab6afdafafce9103
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.902 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |727 |727 |0 |
|test (thor) |621 |621 |0 |
|test (roxie) |748 |748 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: f18d6199b13081737aa6c02c5528e5ea3f55faa5
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.232 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |727 |727 |0 |
|test (thor) |621 |621 |0 |
|test (roxie) |748 |748 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith reviewed (approved):
Looks good.
